### PR TITLE
games-fps/prboom-plus: Bump to version 2.5.1.5.4540.1

### PR DESCRIPTION
**Pull request by vilhelmgray**
The non-free USE flag is removed because non-free assets are no longer
provided in the source tarball. The png USE flag is removed since
sdl2-image provides that functionality now. The sdl-image and sdl-mixer
USE flags are renamed to sdl2-image and sdl2-mixer respectively to match
the use of the matching SDL2 libraries now.

Closes: https://bugs.gentoo.org/703610
Package-Manager: Portage-2.3.84, Repoman-2.3.20
Signed-off-by: William Breathitt Gray <vilhelm.gray@gmail.com>

### CODE REVIEW

File: games-fps/prboom-plus/metadata.xml [hunk 1]
```diff
@@ -2,7 +2,9 @@
 <!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
 	<longdescription lang="en">
-		PrBoom+ is a Doom source port developed from the original PrBoom project.
+		PrBoom+ is an enhanced source port of the classic 3D first-person shooter game Doom, first released by id Software in 1993. It uses the Simple Direct Media layer (SDL) library and features an optional OpenGL renderer. It is based on PrBoom, MBF and LxDoom, which in turn are based on TeamTNT's Boom, a freely available port of Doom for DOS.
+
+		In addition to PrBoom's features, PrBoom+ offers uncapped framerate, variable gamespeed, re-record, walkcam, chasecam, full mouselook, FOV, and other features without loss of compatibility with the original Doom.
```
Comment (ulm) [inline games-fps/prboom-plus/metadata.xml]: Could you please wrap these long lines to less than 80 characters length?

File: games-fps/prboom-plus/prboom-plus-2.5.1.5.4540.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+inherit autotools desktop xdg
+
+MY_PV=$(ver_rs 4 "+svn" 5 "+dfsg")
+MY_P=${PN}-${MY_PV}
+DESCRIPTION="An enhanced clone of the classic first-person shooter Doom"
+HOMEPAGE="https://prboom-plus.sourceforge.net"
+SRC_URI="http://deb.debian.org/debian/pool/main/p/prboom-plus/${PN}_${MY_PV}.orig.tar.xz -> ${P}.tar.xz"
+
+LICENSE="BSD BSD-with-disclosure GPL-2+ LGPL-2.1+ MIT public-domain"
```
Comment (ulm) [inline games-fps/prboom-plus/prboom-plus-2.5.1.5.4540.1.ebuild]: GPL-2+ would still be the main license, right? So it should be listed first.

File: games-fps/prboom-plus/prboom-plus-2.5.1.5.4540.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	net? ( media-libs/sdl2-net )
+	pcre? ( dev-libs/libpcre:3 )
+	portmidi? ( media-libs/portmidi )
+	sdl2-image? ( media-libs/sdl2-image )
+	sdl2-mixer? ( media-libs/sdl2-mixer[midi] )
+	vorbis? ( media-libs/libvorbis )"
+RDEPEND="${DEPEND}"
+
+S=${WORKDIR}/${MY_P}
+PATCHES="${FILESDIR}/${PN}-2.5.1.4-Remove-nonstandard-gamesdir-variable.patch"
```
Comment (ulm) [inline games-fps/prboom-plus/prboom-plus-2.5.1.5.4540.1.ebuild]: `PATCHES` should be an array, even if it has only one element.

```suggestion
PATCHES=( "${FILESDIR}/${PN}-2.5.1.4-Remove-nonstandard-gamesdir-variable.patch" )
```

File: games-fps/prboom-plus/prboom-plus-2.5.1.5.4540.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+src_prepare(){
```
Comment (ulm) [inline games-fps/prboom-plus/prboom-plus-2.5.1.5.4540.1.ebuild]: This looks strange without a space before the `{`.

File: games-fps/prboom-plus/prboom-plus-2.5.1.5.4540.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+		$(use_with net) \
+		$(use_with pcre) \
+		$(use_with portmidi) \
+		$(use_with sdl2-image image) \
+		$(use_with sdl2-mixer mixer) \
+		$(use_with vorbis vorbisfile) \
+		--with-waddir="${EPREFIX}/usr/share/doom"
+}
+
+src_install(){
```
Comment (ulm) [inline games-fps/prboom-plus/prboom-plus-2.5.1.5.4540.1.ebuild]: Ditto.

File: profiles/license_groups [hunk 6]
```diff
@@ -30,7 +30,7 @@ OSI-APPROVED AFL-3.0 AGPL-3 AGPL-3+ Apache-1.1 Apache-2.0 APL-1.0 APSL-2 Artisti
...
+MISC-FREE AIFFWriter.m Allegro alternate AMPAS bea.ri.jsr173 BEER-WARE boehm-gc BSD-1 BSD-with-attribution BSD-with-disclosure buddy bufexplorer.vim BZIP2 canfep CAOSL CDDL-1.1 CDDL-Schily CeCILL-C CMake CPL-0.5 CRACKLIB Crypt-IDEA DES docbook dom4j DUMB-0.9.3 eGenixPublic-1.1 ElementTree Emacs ErlPL-1.1 FastCGI feh File-MMagic Flashpix FLEX flexmock FLTK freetts FVWM gd gsm HTML-Tidy htmlc iASL icu IDPL imagemagick Info-ZIP inner-net Interbase-1.0 ipadic ipx-utils Ispell JDOM JNIC JOVE Khronos-CLHPP LambdaMOO LIBGLOSS libmng libpng libpng2 libtiff LLVM-Grant LPPL-1.3 LPPL-1.3b lsof Mail-Sendmail matplotlib Mini-XML minpack MIT-with-advertising mm mpich2 NCSA-HDF netcat NEWLIB ngrep Old-MIT openafs-krb5-a Openwall otter par PCRE perforce photopc PHP-2.02 pngcrush pngnq Princeton psutils rc rdisc regexp-UofT repoze RSA rwpng sash scanlogd sdlsasteroids Sendmail Sendmail-Open-Source shrimp SMAIL Snd SNIA SSLeay Subversion SVFL symlinks tablelist tcltk tcp_wrappers_license TeX TeX-other-free the-Click-license Time-Format Time-modules tm-align torque-2.5 totd Toyoda UCAR-Unidata URT VTK w3m wm2 x2x xbatt xboing XC Xdebug xtrs xvt ZSH
```
Comment (ulm) [inline profiles/license_groups]: :+1: I agree that this qualifies as a free software license.

**Comment by vilhelmgray** [comment]
> Please confirm that GPL-2+ is the main license, and I can fix the other small issues when committing.
> (Alternatively, feel free to push an update.)

Yes the main license GPL-2+, with the other licenses being from code pulled in from other projects I believe. The respective prboom-plus Debian package conveniently provides an encompassing list of the copyrights found in the source tarball: https://metadata.ftp-master.debian.org/changelogs//main/p/prboom-plus/prboom-plus_2.5.1.5+svn4540+dfsg1-1_copyright

Since these are minor issues, please go ahead and fix them when you commit. Thank you. :-)

**Comment by ulm** [comment]
Merged, with the discussed updates applied.

**Pull request closed without merging**